config/clusters/callysto/common.values.yaml

```diff
@@ -9,6 +9,14 @@ nfs:
     # Name of Google Filestore share
     baseShareName: /homes/
 jupyterhub:
+  # pre-puller is necessary as the image is pretty big, and
+  # pulling during first user spawn might cause timeouts and poor user
+  # experience. Also helps with node pre-warming.
+  prePuller:
+    continuous:
+      enabled: true
+    hook:
+      enabled: true
```

Comment:
Pre-puller is a helm chart thing that is decoupled from the configurator, due to that, only whats configured in the helm chaft actually helps here.

If there is a mismatch, this is slowing things down instead, because nodes will pull both this helm chart managed image, and then on demand the actual image.

I think a note to clarify this is a judgement call concluded to be positive in this case, because some motivation.

Reply:
Thank you, I think I understand. To update the image, I update `singleuser.image.name` and `singleuser.image.tag` here, _**and**_ the corresponding values in the configurator. That way, when a new node is added, the image is `prePulled` before any users arrive. When they do, they'll be given whatever image is defined in the configurator, so we want to make sure this is the same image which was prePulled, correct?

Right now,  when I go to the configurator I get a 403, but I can sort that out with support as long as I have the right idea above. If that is the case, please go ahead and merge this.

Reply:
No need to change the configurator, just update the image in this config _and leave the configurator blank_. It will default to pulling the image here.